### PR TITLE
add dealing-with-bad-data example

### LINKED ISSUE #6: Pattern: Dealing with bad data
https://cloud.google.com/blog/big-data/2017/06/guide-to-common-cloud-dataflow-use-case-patterns-part-1
https://cloud.google.com/blog/big-data/2016/01/handling-invalid-inputs-in-dataflow

### PR DESCRIPTION
**Pull request by Kenji-H**
### Issues
resolve #6 

### TODOs
- [x] README
- [x] Add to Summary Table

### CODE REVIEW

File: dead-letter/pom.xml [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+
+    <dependency>
+      <groupId>org.mockito</groupId>
+      <artifactId>mockito-core</artifactId>
+      <version>${mockito.version}</version>
+      <scope>test</scope>
+    </dependency>
+  </dependencies>
+
+</project>
```
Comment (byam) [inline dead-letter/pom.xml]: It seems most of these libraries are not used. Could you remove unused dependencies?

**Review by byam — CHANGES_REQUESTED** [review]
pom.xml is too large. Others are good to me.

**Comment by Kenji-H** [comment]
@byam Thank you for your review. I made some changes based on your comments.

**Review by byam — APPROVED** [review]
LGTM!

**Pull request merged**